packages/clerk-js/src/ui/utils/phoneUtils.ts

```diff
@@ -115,3 +116,31 @@ export function stringToFormattedPhoneString(str: string): string {
   const parsed = parsePhoneString(str);
   return `+${parsed.code} ${formatPhoneNumber(parsed.number, parsed.pattern, parsed.code)}`;
 }
+
+type CountryData = {
+  searchTerm: string;
+  country: CountryEntry;
+};
+
+const priorityCodes: string[] = ['1'];
+export function getLongestValidCountryCode(value: string, options: CountryData[]) {
+  let result = '';
+  let selectedCountryCode = null;
+  const numberValue = value.replace('+', '');
+
+  for (let i = 4; i > 0; i--) {
+    const code = numberValue.slice(0, i);
+    const priority = numberValue.slice(0, 1);
+    const hasPriority = priorityCodes.includes(priority);
+
+    result = numberValue.slice(hasPriority ? 1 : i, value.length);
+    selectedCountryCode = options.find(o => o.country.code === (hasPriority ? priority : code));
+
+    if (selectedCountryCode) {
```

Comment:
If a selected code is found, instead of using `break` you can simply return early 🙃